ci/Makefile

```diff
@@ -49,30 +73,27 @@ restart-batch-proxy:
                            -o name \
                          | sed 's:pods/::' \
                          | head -n 1))
-	kubectl port-forward ${BATCH_POD} 8888:5000 > batch-proxy.log 2>batch-proxy.err & echo $$! > batch-proxy.pid
-
-
+	kubectl port-forward $(BATCH_POD) 8888:5000 > batch-proxy.log 2>batch-proxy.err & echo $$! > batch-proxy.pid
 
 run-local: HAIL_CI_REMOTE_PORT = 3000
-run-local: PROXY_IP=$(shell gcloud compute instances list --filter=name=dk-test --format="value(EXTERNAL_IP)")
+# SELF_HOSTNAME and BATCH_SERVER_URL assignment run in shell context
+# and therefore cannot have a space (else it's a 2 operand command)
```

Comment:
I don't think this comment is necessary. We should assume Makefile syntax fluency of our developers and ensure everyone is educated on it.

Reply:
Upon reflection, this comment is more curt than I had intended.

This knowledge is clearly crucial to understanding how this particular make target works. It is, however, knowledge generally necessary to understanding all of our make files. I want to avoid explicitly stating the syntax and semantics of make on many lines of our various Makefiles. I think an economical solution is to assume a common base of knowledge among our developers *and*, crucially, to encode that knowledge somewhere.

Make is increasingly central to our developer operations. As our organization grows (John C may join us soon, the co-op Daniel G starts in January) we need a way to disseminate this knowledge efficiently. Maybe we should add a separate dev-docs folder to the repo that contains short READMEs that describe the basic information, with links to further resources, that are necessary for understanding our codebase. Some of these might be as simple as "here's a tutorial on using Make". We may also find that existing resources are unsatisfactory, leading us to add our own information. If someone does a hail lab meeting about a topic, the resources of that topic should go here.

cc: hail-team (can we make GitHub aliases?), @tpoterba @cseed @patrick-schultz @jigold @catoverdrive @jbloom22 @chrisvittal

Reply:
Aside, perhaps it's worthwhile for me to run a hail lab meeting on GNU Make. I think I'd only need half an hour.

Reply:
Yeah I agree with this, and thanks for amending your response. I’m generally in favor of comments, because in practice I’ve found that people think differently enough from one another that they help understanding more than they hurt readability. I do appreciate that views differ on this. 

It definitely makes sense to centralize make documentation. I’ll remove the comment.